### DISCUSSION TITLE
Micropython-OSError: -202-mip.install('sensor')

### DISCUSSION THREAD

**Discussion by awchen9** [micropython/micropython#13603] [Libraries & Drivers]
>>> import mip
>>> mip.install('sensor')
Installing sensor (latest) from https://micropython.org/pi/v2 to /lib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mip/__init__.py", line 1, in install
  File "mip/__init__.py", line 1, in _install_package
  File "mip/__init__.py", line 1, in _install_json
  File "requests/__init__.py", line 180, in get
  File "requests/__init__.py", line 76, in request
OSError: -202
>>>

**Comment by Josverl** [comment]
**OSError: -202** is an error code that indicates a failure in the getaddrinfo() function, which is used to resolve a hostname to an IP address. 
This error can (will) occur when the hostname is invalid, the network is unreachable, or the DNS server is not responding.

**Reply by awchen9** [reply]
I am using the "esp32-cam" development board and developing it using Tonny. Is there any solution

**Reply by mcauser** [reply]
Maybe we should add a list of the common ones to: https://docs.micropython.org/en/latest/library/errno.html

Microbit has one: https://support.microbit.org/support/solutions/articles/19000126750-python-oserror-codes

**Reply by awchen9** [reply]
After connecting to WiFi, run 
>>> import mip
>>> mip.install('sensor')
Installing sensor (latest) from https://micropython.org/pi/v2 to /lib
Package not found: https://micropython.org/pi/v2/package/6/sensor/latest.json
Package may be partially installed
>>> 
Is the URL ' https://micropython.org/pi/v2' or 'https://micropython.org/pi/v2/package/6/sensor/latest.json' incorrect.
How to solve it

**Reply by glenn20** [reply]
First, you should check and verify that your internet access on your micropython board is working. If that is working, the next thing to check is that your DNS settings are correct so that hostname lookups are working.

Note that if you dont have internet access from your micropython device, you can still `mip install` packages using [`mpremote mip install`](https://docs.micropython.org/en/latest/reference/mpremote.html#mpremote-command-mip) fron your computer (if your micropython device is attached to your computer via usb/serial).

**Reply by awchen9** [reply]
![2024-02-27 082001](https://github.com/micropython/micropython/assets/159148933/b853c996-b1c1-4b52-bc50-d93025de72ed)
Is it like this

**Reply by bixb922** [reply]
There is no module called "sensor" in the Micropython library. What module or driver are you trying to install?

The command ```mpremote mip install``` works at the command prompt of your PC, and not at the >>> prompt. Please try on your PC. ```mpremote``` is a useful utility program, see https://docs.micropython.org/en/latest/reference/mpremote.html

**Comment by chris35469** [comment]
This is not an actual solution. However, I resolved this error on the Xiao ESP32-S3 board by downgrading the firmware to the following version:

v1.22.0 (2023-12-27)

Link to firmware: https://micropython.org/download/ESP32_GENERIC_S3/

I hope this helps folks running into this error. It took me too long to find a temporary fix for this issue. 

The following thread led me to try this approach.
 
https://forum.micropython.org/viewtopic.php?f=18&t=12543

Hopefully, there will be another fix soon that doesn't require downgrading the firmware.
